samcli/commands/build/command.py

```diff
@@ -0,0 +1,73 @@
+"""
+CLI command for "build" command
+"""
+
+import logging
+import click
+
+from samcli.yamlhelper import yaml_dump
+from samcli.cli.main import pass_context, common_options as cli_framework_options, aws_creds_options
+from samcli.commands._utils.options import template_common_option as template_option
+from samcli.commands.build.build_context import BuildContext
+from samcli.lib.build.app_builder import ApplicationBuilder
+
+LOG = logging.getLogger(__name__)
+
+HELP_TEXT = """
+Use this command to build your Lambda function source code and generate artifacts that can be deployed to AWS Lambda 
+"""
+
+
+@click.command("build", help=HELP_TEXT, short_help="Build your Lambda function code")
+@click.option('--build-dir', '-b',
+              default="build",
+              type=click.Path(),
+              help="Path to a folder where the built artifacts will be stored")
+@click.option("--source-root", "-s",
+              default=".",
+              type=click.Path(),
```

Comment:
`click.Path()` by default allows files and directories. Do we want to constrain this or the `--build-dir` to a directory? Should we have click resolve the path as well? What about the writable or readable checks?

https://click.palletsprojects.com/en/6.x/api/#click.Path

Reply:
Perfect. Made the change